Gifski/AppDelegate.swift

```diff
@@ -8,83 +8,33 @@ extension NSColor {
 
 @NSApplicationMain
 final class AppDelegate: NSObject, NSApplicationDelegate {
-	@IBOutlet private weak var window: NSWindow!
-
-	private var progressObserver: NSKeyValueObservation?
-
-	lazy var circularProgress = with(CircularProgressView(frame: CGRect(widthHeight: 160))) {
-		$0.foreground = .appTheme
-		$0.strokeWidth = 2
-		$0.percentLabelLayer.setAutomaticContentsScale()
-		$0.percentLabelLayer.implicitAnimations = false
-		$0.layer?.backgroundColor = .clear
-		$0.isHidden = true
-		$0.centerInWindow(window)
-	}
-
-	lazy var videoDropView = with(VideoDropView()) {
-		$0.frame = window.contentView!.frame
-		$0.dropText = "Drop a Video to Convert to GIF"
-		$0.onComplete = { url in
-			self.convert(url.first!)
-		}
-	}
+	var mainWindowController: MainWindowController!
 
 	var hasFinishedLaunching = false
 	var urlsToConvertOnLaunch: URL!
-	var choosenDimensions: CGSize!
-	var choosenFrameRate: Int!
-
-	@objc dynamic var isRunning: Bool = false {
-		didSet {
-			if isRunning {
-				videoDropView.isHidden = true
-			} else {
-				videoDropView.fadeIn()
-			}
-
-			circularProgress.isHidden = !isRunning
-		}
-	}
 
 	func applicationWillFinishLaunching(_ notification: Notification) {
 		defaults.register(defaults: [
 			"NSApplicationCrashOnExceptions": true,
 			"NSFullScreenMenuItemEverywhere": false,
 			"outputQuality": 1
 		])
-
-		with(window!) {
-			$0.titleVisibility = .hidden
-			$0.appearance = NSAppearance(named: .vibrantDark)
-			$0.tabbingMode = .disallowed
-			$0.titlebarAppearsTransparent = true
-			$0.isMovableByWindowBackground = true
-			$0.styleMask.remove([.resizable, .fullScreen])
-			$0.styleMask.insert(.fullSizeContentView)
-			$0.isRestorable = false
-			$0.setFrame(CGRect(width: 360, height: 240), display: true)
-			$0.center()
-		}
 	}
 
 	func applicationDidFinishLaunching(_ notification: Notification) {
+		mainWindowController = MainWindowController(windowNibName: NSNib.Name(rawValue: "MainWindowController"))
```

Comment:
The recommended practice is to create a `NSNib` extension top-level and define the name there, so you can just use `.mainWindowController` or something here. That's why they introduced strongly typed names for these APIs.

Reply:
👍 Much better